Make python_agnostic cpp extension tests standalone

jbschlosser opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #153855
* __->__ #153274
* #153264

Related: #148920

This PR:
* Introduces a new file `test/cpp_extensions/python_agnostic_extension/test/test_python_agnostic.py` with testing that follows the usual python testing patterns
    * This replaces the testing for python_agnostic in `test/test_cpp_extensions_aot.py`

After this PR, it is now possible to run:
```
python test/cpp_extensions/python_agnostic_extension/test/test_python_agnostic.py
```

and the test will build the prerequisite wheel before running the tests.

jbschlosser committed:
Make python_agnostic cpp extension tests standalone

[ghstack-poisoned]

jbschlosser committed:
Update on "Make python_agnostic cpp extension tests standalone"




[ghstack-poisoned]

janeyx99 reviewed (commented):
very nice! can we also get a link here from CI to make sure this does indeed run?

janeyx99 reviewed (approved):
thank u! <3

jbschlosser committed:
Update on "Make python_agnostic cpp extension tests standalone"


Related: #148920

This PR:
* Introduces a new file `test/cpp_extensions/python_agnostic_extension/test/test_python_agnostic.py` with testing that follows the usual python testing patterns
    * This replaces the testing for python_agnostic in `test/test_cpp_extensions_aot.py`

After this PR, it is now possible to run:
```
python test/cpp_extensions/python_agnostic_extension/test/test_python_agnostic.py
```

and the test will build the prerequisite wheel before running the tests.

[ghstack-poisoned]

cyyever commented:
@pytorchmergebot merge -i

jbschlosser committed:
Update on "Make python_agnostic cpp extension tests standalone"


Related: #148920

This PR:
* Introduces a new file `test/cpp_extensions/python_agnostic_extension/test/test_python_agnostic.py` with testing that follows the usual python testing patterns
    * This replaces the testing for python_agnostic in `test/test_cpp_extensions_aot.py`

After this PR, it is now possible to run:
```
python test/cpp_extensions/python_agnostic_extension/test/test_python_agnostic.py
```

and the test will build the prerequisite wheel before running the tests.

[ghstack-poisoned]

jbschlosser committed:
Update on "Make python_agnostic cpp extension tests standalone"


Related: #148920

This PR:
* Introduces a new file `test/cpp_extensions/python_agnostic_extension/test/test_python_agnostic.py` with testing that follows the usual python testing patterns
    * This replaces the testing for python_agnostic in `test/test_cpp_extensions_aot.py`

After this PR, it is now possible to run:
```
python test/cpp_extensions/python_agnostic_extension/test/test_python_agnostic.py
```

and the test will build the prerequisite wheel before running the tests.

[ghstack-poisoned]

jbschlosser committed:
Update on "Make python_agnostic cpp extension tests standalone"


Related: #148920

This PR:
* Introduces a new file `test/cpp_extensions/python_agnostic_extension/test/test_python_agnostic.py` with testing that follows the usual python testing patterns
    * This replaces the testing for python_agnostic in `test/test_cpp_extensions_aot.py`

After this PR, it is now possible to run:
```
python test/cpp_extensions/python_agnostic_extension/test/test_python_agnostic.py
```

and the test will build the prerequisite wheel before running the tests.

[ghstack-poisoned]

jbschlosser commented:
@pytorchbot merge
